core/include/cubos/core/al/audio_context.hpp

```diff
@@ -0,0 +1,200 @@
+        /// @param[out] devices Vector to fill with the available devices.
+        virtual void enumerateDevices(std::vector<std::string>& devices) = 0;
+
+        /// @brief Creates a new audio device
+        /// @param specifier The specifier of the audio device, used to identify it
+        /// @param listenerCount The number of audio listener to be created by the device's engine.
+        /// @return Handle of the new device
+        virtual AudioDevice createDevice(unsigned int listenerCount,
+                                         const std::string& specifier = "MiniaudioDevice") = 0;
+
+        /// @brief Creates a new audio buffer.
+        /// @param data Data to be written to the buffer, either .wav, .mp3 or .flac.
+        /// @param datSize Size of the data to be written.
+        /// @return Handle of the new buffer.
+        virtual Buffer createBuffer(const void* data, size_t dataSize) = 0;
+    };
+
+    /// @brief Namespace to store the abstract types implemented by the audio device implementations.
+    namespace impl
+    {
+        /// @brief Abstract audio buffer.
+        class CUBOS_CORE_API Buffer
+        {
+        public:
+            virtual ~Buffer() = default;
+
+            /// @brief  Gets the length in seconds of the audio buffer.
+            /// @return Length in seconds of the audio buffer.
+            virtual size_t length() = 0;
+
+        protected:
+            Buffer() = default;
+        };
+
+        /// @brief Abstract audio source.
+        class CUBOS_CORE_API Source
+        {
+        public:
+            virtual ~Source() = default;
+
+            /// @brief Sets the buffer to be played by the source.
+            /// @param buffer Buffer.
+            virtual void setBuffer(cubos::core::al::Buffer buffer) = 0;
+
+            /// @brief Sets the position of the source, by default, in the world space.
+            /// @see setRelative() to change this behavior.
+            /// @param position Position.
+            virtual void setPosition(const glm::vec3& position) = 0;
+
+            /// @brief Sets the velocity of the source, by default, in the world space.
+            /// @param velocity Velocity.
+            virtual void setVelocity(const glm::vec3& velocity) = 0;
+
+            /// @brief Sets the gain of the source.
+            /// @param gain Gain.
+            virtual void setGain(float gain) = 0;
+
+            /// @brief Sets the pitch of the source.
+            /// @param pitch Pitch.
+            virtual void setPitch(float pitch) = 0;
+
+            /// @brief Sets whether the source plays in a loop.
+            /// @param looping Looping flag.
+            virtual void setLooping(bool looping) = 0;
+
+            /// @brief Sets whether the source position and velocity is relative to the listener or
+            /// not.
+            /// @param relative Relative flag.
+            virtual void setRelative(bool relative) = 0;
```

Comment:
For this to make sense, the source should be pinned to a specific listener right?
Maybe we can pass it here as an argument, and then use `ma_sound_set_pinned_listener_index` in the implementation?